LeanOA/ForMathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Abs.lean

```diff
@@ -0,0 +1,236 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Order
+
+/-!
+# Absolute value of an operator defined via the continuous functional calculus
+
+This file defines the absolute value via the (unital and non unital) continuous functional calculus
+(CFC) and (CFCₙ), and includes foundational API.
+
+## Main declarations
+
++ `CFC.abs`: The absolute value declaration as `abs a := sqrt (star a) * a`.
+
+# TODO
+
+There is likely an `RCLike` version of `abs_smul_complex`.
+
+-/
+
+open scoped NNReal
+
+namespace CFC
+
+section Generic
+
+variable {A : Type*}
+
+section NonUnital
+
+section Real
+
+variable [NonUnitalRing A] [StarRing A] [TopologicalSpace A]
+variable [PartialOrder A] [StarOrderedRing A] [Module ℝ A] [SMulCommClass ℝ A A] [IsScalarTower ℝ A A]
+variable [NonUnitalContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
+variable [NonnegSpectrumClass ℝ A]
+
+/-- The absolute value of an operator, using the nonunital continuous functional calculus. -/
+noncomputable def abs (a : A) := sqrt (star a * a)
+
+@[simp]
+lemma abs_neg (a : A) : abs (-a) = abs a := by
+  simp [abs]
+
+@[simp]
+lemma abs_nonneg {a : A} : 0 ≤ abs a := sqrt_nonneg
+
+lemma abs_star {a : A} (ha : IsStarNormal a) : abs (star a) = abs a := by
+  rw [abs, abs, star_comm_self, star_star]
+
+@[simp]
+lemma abs_zero : abs (0 : A) = 0 := by
+  rw [abs, star_zero, mul_zero, sqrt_zero]
+
+variable [IsTopologicalRing A] [T2Space A]
+
+lemma abs_mul_self (a : A) : (abs a) * (abs a) = star a * a := by
+  refine sqrt_mul_sqrt_self _ <| star_mul_self_nonneg _
+
+lemma abs_nnrpow_two (a : A) : (abs a) ^ (2 : NNReal) = star a * a := by
```

Comment:
Although, to be honest, I don't really like this lemma (i.e., I don't like that `2 : ℝ≥0`, but I realize we're in a non-unital algebra).
```suggestion
lemma abs_nnrpow_two (a : A) : abs a ^ (2 : ℝ≥0) = star a * a := by
```